quarkchain/cluster/cluster_config.py

```diff
@@ -455,7 +497,10 @@ def from_dict(cls, d):
         config.QUARKCHAIN = QuarkChainConfig.from_dict(config.QUARKCHAIN)
         config.MONITORING = MonitoringConfig.from_dict(config.MONITORING)
         config.MASTER = MasterConfig.from_dict(config.MASTER)
-        config.SLAVE_LIST = [SlaveConfig.from_dict(s) for s in config.SLAVE_LIST]
+        config.SLAVE_LIST = [
+            SlaveConfig.from_dict(s, config.QUARKCHAIN.CHAINS)
```

Comment:
Oh, I think I may point to incorrect config variable (although CHAINS would work), which should be QUARKCHAIN.CHAINS_SIZE

Reply:
since the backward-compatible conversion only works if shard size per chain is 1, I added an enforcing check as follows 

`check(all(chain.SHARD_SIZE == 1 for chain in chains))`

which needs the whole `QUARKCHAIN.CHAINS` config